### PR TITLE
:sparkles: (dmk) [DSDK-703]: Implement GetDeviceMetadata device action

### PR DESCRIPTION
**Pull request by paoun-ledger**
### 📝 Description

Part of ConnectApp implementation as defined here:
https://ledgerhq.atlassian.net/wiki/spaces/WXP/pages/5674729959/OS+Get+Device+Metadata

ticket:
https://ledgerhq.atlassian.net/browse/DSDK-703

### ❓ Context

- **JIRA or GitHub link**:

- **Feature**:

### ✅ Checklist

Pull Requests must pass CI checks and undergo code review. Set the PR as Draft if it is not yet ready for review.

- [ ] **Covered by automatic tests** 
- [ ] **Changeset is provided** 
- [ ] **Documentation is up-to-date** 
- [ ] **Impact of the changes:** 
  - _list of the changes_

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.

### CODE REVIEW

File: packages/device-management-kit/src/api/command/os/ListLanguagePackCommand.test.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import { CommandResultFactory } from "@api/command/model/CommandResult";
+import { isSuccessCommandResult } from "@api/command/model/CommandResult";
+import { ListLanguagePackCommand } from "@api/command/os/ListLanguagePackCommand";
+import { ApduResponse } from "@api/device-session/ApduResponse";
+import { UnknownDeviceExchangeError } from "@root/src";
```
Comment (jdabbech-ledger) [inline packages/device-management-kit/src/api/command/os/ListLanguagePackCommand.test.ts:5]: [should] import from correct file

File: packages/device-management-kit/src/api/command/os/GetCustomImageSizeCommand.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+} from "@api/command/model/CommandResult";
+import {
+  type CommandErrors,
+  isCommandErrorCode,
+} from "@api/command/utils/CommandErrors";
+import { CommandUtils } from "@api/command/utils/CommandUtils";
+import { GlobalCommandErrorHandler } from "@api/command/utils/GlobalCommandError";
+import { type ApduResponse } from "@api/device-session/ApduResponse";
+import { type CommandErrorArgs, DeviceExchangeError } from "@api/Error";
+import { UnknownDeviceExchangeError } from "@root/src";
```
Comment (jdabbech-ledger) [inline packages/device-management-kit/src/api/command/os/GetCustomImageSizeCommand.ts:17]: import 👀

File: packages/device-management-kit/src/api/command/os/GetCustomImageSizeCommand.test.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import { CommandResultFactory } from "@api/command/model/CommandResult";
+import { isSuccessCommandResult } from "@api/command/model/CommandResult";
+import { GetCustomImageSizeCommand } from "@api/command/os/GetCustomImageSizeCommand";
+import { ApduResponse } from "@api/device-session/ApduResponse";
+import { UnknownDeviceExchangeError } from "@root/src";
```
Comment (jdabbech-ledger) [inline packages/device-management-kit/src/api/command/os/GetCustomImageSizeCommand.test.ts:5]: same

File: packages/device-management-kit/src/api/command/os/ListLanguagePackCommand.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            errorCode,
+          }),
+        });
+      }
+      return CommandResultFactory({
+        error: GlobalCommandErrorHandler.handle(apduResponse),
+      });
+    }
+
+    const tlvVersion = parser.extract8BitUInt();
```
Comment (jdabbech-ledger) [inline packages/device-management-kit/src/api/command/os/ListLanguagePackCommand.ts:90]: [ask] does `tlv` means total length value?

File: packages/device-management-kit/src/internal/manager-api/data/AxiosManagerApiDataSource.ts [hunk 5]
```diff
@@ -113,11 +207,34 @@ export class AxiosManagerApiDataSource implements ManagerApiDataSource {
...
+        },
+      ),
+    )
+      .map((res) => res.data)
+      .chain((apps) => this.mapLanguagesDtoToLanguages(apps))
       .mapLeft((error) => new HttpFetchApiError(error));
   }
 
-  private mapAppTypeDtoToAppType(appType: AppTypeDto): AppType {
+  private mapAppTypeDtoToAppType(appType: AppTypeDto | null): AppType | null {
```
Comment (jiyuzhuang) [inline packages/device-management-kit/src/internal/manager-api/data/AxiosManagerApiDataSource.ts:234]: _[COULD]_ I think that all the `map*` functions are _STATIC_ and they take a lot of spaces in this file, so perhaps putting them in a separate utility file is a good idea.

File: packages/device-management-kit/src/internal/manager-api/data/ManagerApiDataSource.ts [hunk 6]
```diff
@@ -66,4 +73,51 @@ export interface ManagerApiDataSource {
...
+
+  /**
+   * Retrieves the available language packages for a device.
+   *
+   * @param params - The parameters for getting the language packages.
+   * @returns EitherAsync containing the list of language packages or an HttpFetchApiError.
+   */
+  getLanguagePackages(
+    params: GetLanguagePackagesParams,
+  ): EitherAsync<HttpFetchApiError, LanguagePackage[]>;
```
Comment (jiyuzhuang) [inline packages/device-management-kit/src/internal/manager-api/data/ManagerApiDataSource.ts:115]: _[COULD]_ here and below
```suggestion
  ): EitherAsync<HttpFetchApiError, Array<LanguagePackage>>;
```

File: packages/device-management-kit/src/api/secure-channel/device-action/InstallApp/InstallAppDeviceAction.ts [hunk 7]
```diff
@@ -139,6 +139,18 @@ export class InstallAppDeviceAction extends XStateDeviceAction<
...
...3uoVjaNClhPqIL5vh+qq2JY1KBO+GgaBmK5hIBBY8iB+5sCaYAYAA8rAABqNZyCwbqoMUpTlFU3Z0YxLFOmxqDBkOoYISo6hsrMWhBL4OiiNCixqEm3xLsYWjWIYWgZv0mzkTulGHPyNH8cxrHJNBomXmS3SjpoqysPGL7fDoUJQgYOhJp5S7Mssr76KIfj-NuuxGXyppmYJwlWbBYnwXZiEIA5BhORhIQuG57meapiysACinBMM4IzFuBlhcBxmRRguSnBgHFcZ6vEwDVxB1dZMoJfKLgQvlsb+GhOg2Es3npmCXjGEsIxEXC5UxOFoGma1hB1VZF6dbZ3WMno+WaEFIREQELiqXhrAEWYxFQrYOhkdshmVRFJZ0bVtSxdKw7XpJCBjFoTkBTYjKmENc6qjCrDEY+Gj6E+-lPqF80PYtrC0ctq33BKwhSnBm2jplaiDLYWhmHojIaAMKmqgYmj5cYFgWKY6rERo+l3RV+xUSZyPPW1r3owStjveJiVfXjBNE8RpPk0mbLUhsG4M8sOjwwa7NVeiQGEA1Ho8Y192EB1H0SZSQSsP8gKTFDO3Mn+Oq5qzCOq49Aoa4INZ1g2TYtu2TpdqUesG0L8opgTKqTGmr4-iykK22VeaoEwEBwEoGtFtR2MjklAC0OWqpn0LLgsAwrIYaZ2Mru4c8cRY8Hwkw2RnX1rEmbiDMyaYAnt7jXeXC3UYKdQFji8DxTjSWiEmFipd4Ufahys32yrqec3VWJD2e6efb04+qu+re-qy-52-metL6a4F9ieG9G64Srg6YGaWHYN2iDd87qpHsbLKs6wDD3iN94eCC-ZTx8yvsLXoN0XB33fHYf4hMOQnR2jdf4egQQvnTMsP+jskYowEhZCShtwHqDcPhIKlgdD9BGJ5KwqkXA7WLt4FYbgrCLDKgvCuasDxRXwagdeI8G63kZNSdBgUUzTChObL4j8nILm-oTK6SwsGnyeqjWoYDurph2mMQqj4hruGOjhNwYI4ELHVAuYiSi9ycxRi9TAfD66b1cJNbQpivCeXjERSRCBUGzGUjCTKj8oSKUsZXdWFFiDqNHNvMOiltDTxtnPEJnC2AAGUACu1BaCwGHg46+30zDaGsF-EElhSkjCTCCME6wobOSMLdY+bNlECgAKK1nrJEpKmVfpFJWCUspDgcLBBpnTDCZhliLn0uEIAA */
```
Comment (jiyuzhuang) [inline packages/device-management-kit/src/api/secure-channel/device-action/InstallApp/InstallAppDeviceAction.ts:156]: [ASK] I'm curious about the layout here, the action has been changed, so the diagram should be changed too.

Comment (paoun-ledger) [reply packages/device-management-kit/src/api/secure-channel/device-action/InstallApp/InstallAppDeviceAction.ts:156]: actually I did the modification in vim so no vscode extension there :sob: 
But when looking at it in a xstate visualizer it don't change the layout much, there is just now a subtext in the already existing onDone transition

File: packages/device-management-kit/src/api/command/os/GetCustomImageSizeCommand.ts [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+      }
+      return CommandResultFactory({
+        error: GlobalCommandErrorHandler.handle(apduResponse),
+      });
+    }
+
+    const size = parser.extract32BitUInt();
+    if (size === undefined) {
+      return CommandResultFactory({
+        error: new UnknownDeviceExchangeError("Didn't receive any size"),
```
Comment (aussedatlo) [inline packages/device-management-kit/src/api/command/os/GetCustomImageSizeCommand.ts:80]: [COULD] We use `InvalidStatusWordError` in all other commands instead of  `UnknownDeviceExchangeError`

File: packages/device-management-kit/src/api/command/os/ListLanguagePackCommand.ts [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+    if (tlvVersion === undefined) {
+      // No more language package
+      return CommandResultFactory({
+        data: null,
```
Comment (aussedatlo) [inline packages/device-management-kit/src/api/command/os/ListLanguagePackCommand.ts:97]: [COULD] use `undefined` for more consistency with other commands that return nothing

File: packages/device-management-kit/src/api/device-action/task/GetApplicationsMetadataTask.ts [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+      return CommandResultFactory({
+        error: new InvalidStatusWordError("Cannot get the languages catalog"),
+      });
+    }
+  }
+
+  private isApplication(app: InstalledApp): boolean {
+    // Applications with no "code" hash are not real applications
+    // (typically a language package).
+    return app.hashCode != ZERO_HASH;
```
Comment (aussedatlo) [inline packages/device-management-kit/src/api/device-action/task/GetApplicationsMetadataTask.ts:135]: [SHOULD] use `!==`

File: packages/device-management-kit/src/internal/manager-api/data/ManagerApiDto.ts [hunk 11]
```diff
@@ -7,31 +7,123 @@ export enum AppTypeDto {
...
+  device_versions: Id[];
+  providers: Id[];
+  minimum_live_common_version: string | null;
+};
+
+export type FirmwareFinalVersionDto = {
+  id: Id;
+  name: string;
+  version: string;
+  se_firmware: Id | null;
```
Comment (aussedatlo) [inline packages/device-management-kit/src/internal/manager-api/data/ManagerApiDto.ts:78]: [ASK] why null and not undefined?

**Review by aussedatlo — APPROVED** [review]

File: packages/device-management-kit/src/internal/manager-api/data/ManagerApiDto.ts (hunk 11 above)
Comment (paoun-ledger) [reply packages/device-management-kit/src/internal/manager-api/data/ManagerApiDto.ts:78]: it's the dto here, the backend really return `null` when the field is unset
